vfs/filter_streams.cc

```diff
@@ -805,22 +807,13 @@ struct RegisterFileHandler {
                 return UriHandler(cin.rdbuf(), nullptr, info);
             }
 
+            std::string uri = scheme +  "://" + resource;
             Datacratic::FsObjectInfo info
-                = Datacratic::getUriObjectInfo(scheme + "://" + resource);
-
-            if (options.count("mapped")) {
+                = Datacratic::getUriObjectInfo(uri);
 
-                boost::iostreams::mapped_file_source source(resource);
-                shared_ptr<std::streambuf> buf(new boost::iostreams::stream_buffer<boost::iostreams::mapped_file_source>(source));
-                //if (!buf->is_open())
-                //    throw ML::Exception("couldn't open file %s: %s",
-                //                        resource.c_str(), strerror(errno));
-                UriHandlerOptions options;
-                options.mapped = source.data();
-                options.mappedSize = source.size();
-                return UriHandler(buf.get(), buf, info, options);
-            }
-            else {
+            // MLDB-1303 mmap fails on empty files - force filebuf interface
+            // on empty files despite the mapped option
+            if (!options.count("mapped") || !Datacratic::getUriSize(uri)) {
```

Comment:
We should use the size that's in the getUriObjectInfo, to avoid a second call to stat().